tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -530,19 +530,11 @@ describe(__filename, () => {
     );
   });
 
-  it('renders AddonsByAuthorsCard without a user', () => {
+  it('does not render AddonsByAuthorsCards without a user', () => {
```

Comment:
Is the reason this used to be able to render an `AddonsByAuthorsCard` without a user but can no longer because it used to be able to fetch add-ons via the `username` but now it needs an actual user in order to have access to the user's id? If so this makes sense, but it is a change in behaviour, correct? Is it something we're okay with?

Reply:
> Is the reason this used to be able to render an AddonsByAuthorsCard without a user but can no longer because it used to be able to fetch add-ons via the username but now it needs an actual user in order to have access to the user's id? If so this makes sense

Yep.

>  but it is a change in behaviour, correct?

Yep, mostly because it fixes a bug when a user changes their username.

> Is it something we're okay with?

We do not have any other choice currently, but https://github.com/mozilla/addons-frontend/pull/6686 will fix that by adding a "loading state" to the `AddonsByAuthorsCard` component, which enables us to always render this component. But that's the future 😃